### PR TITLE
serializers: remove emails from api response

### PR DESCRIPTION
**Pull request by MJedr**
Ref: #1374

### CODE REVIEW

File: backend/inspirehep/records/marshmallow/authors/base.py [hunk 1]
```diff
@@ -26,6 +27,17 @@ class Meta:
     email_addresses = NonHiddenRaw(dump_only=True)
 
 
+class AuthorsApiSchema(AuthorsRawSchema):
+    class Meta:
+        exclude = AuthorsPublicSchema.Meta.exclude + ["email_addresses"]
+
+    acquisition_source = fields.Method("get_acquisition_source")
```
Comment (drjova) [inline backend/inspirehep/records/marshmallow/authors/base.py:34]: You could add these in ``AuthorsPublicSchema``

File: backend/inspirehep/records/marshmallow/jobs/base.py [hunk 2]
```diff
@@ -21,6 +27,20 @@ def __init__(self, *args, **kwargs):
...
+    reference_letters = fields.Method("get_reference_letters")
+    acquisition_source = fields.Method("get_acquisition_source")
+
+    @staticmethod
+    def get_acquisition_source(data):
+        return get_acquisition_source_without_email(data)
+
+    @staticmethod
+    def get_reference_letters(data):
+        return get_reference_letters_without_email(data)
```
Comment (drjova) [inline backend/inspirehep/records/marshmallow/jobs/base.py:41]: You could add these in `JobsPublicSchema`

File: backend/inspirehep/records/marshmallow/literature/base.py [hunk 3]
```diff
@@ -49,6 +51,19 @@ class Meta:
     report_numbers = NonHiddenRaw(dump_only=True)
 
 
+class LiteratureApiSchema(LiteraturePublicSchema):
```
Comment (drjova) [inline backend/inspirehep/records/marshmallow/literature/base.py:54]: ditto

File: backend/inspirehep/records/serializers/json/jobs.py [hunk 4]
```diff
@@ -17,9 +17,14 @@
     wrap_schema_class_with_metadata(JobsPublicSchema), index_name="records-jobs"
 )
 
+jobs_json_2 = JSONSerializer(
```
Comment (drjova) [inline backend/inspirehep/records/serializers/json/jobs.py:20]: if we add everything on `JobsPublicSchema` don't need to change anything here :)

File: backend/inspirehep/records/marshmallow/authors/base.py (hunk 1 above)
Comment (MJedr) [reply backend/inspirehep/records/marshmallow/authors/base.py:34]: The problem is that `AuthorsDetailSchema` is inheriting from `AuthorsPublicSchema` and if I add it I'll exclude emails also from the detailed page and we don't want to do it

File: backend/inspirehep/records/marshmallow/literature/base.py (hunk 3 above)
Comment (MJedr) [reply backend/inspirehep/records/marshmallow/literature/base.py:54]: It's the same, I can't add it to PublicSchema because I'll change detailed page

**Comment by drjova** [comment]
@MJedr I see, what if we do it in ListSchema for all collections instead of adding a new Api schema? This will modify only search results.

**Comment by MJedr** [comment]
But we use ListSchemas only for `"application/vnd+inspire.record.ui+json"` and we should remove emails from the response for `"application/json"`. Now we use for that PublicSchemas which have more fields than the BaseSchemas from which inherits ListSchemas so I am nor sure if we can use them for `"application/json"` response.

File: backend/inspirehep/records/marshmallow/authors/base.py (hunk 1 above)
Comment (drjova) [reply backend/inspirehep/records/marshmallow/authors/base.py:34]: I see your point, so let's do the following:

We keep it as you did but rename `CollectionApiSchema` to `CollectionPublicListSchema` (i.e. `AuthorPublicListSchema`)

File: backend/inspirehep/records/serializers/json/literature.py [hunk 5]
```diff
@@ -31,6 +32,16 @@
...
+    [
+        (
+            lambda _: is_superuser_or_cataloger_logged_in(),
+            wrap_schema_class_with_metadata(LiteratureAdminSchema),
+        ),
+        (None, wrap_schema_class_with_metadata(LiteraturePublicListSchema)),
+    ]
+)
+
+literature_json_detailed = ConditionalMultiSchemaJSONSerializer(
```
Comment (drjova) [inline backend/inspirehep/records/serializers/json/literature.py:44]: in some places it's `_detailed` in some others is `_record`, let unify it 

```collection_json``` -> for record response
```collection_json_search``` -> for search response

WDYT?

**Review by drjova — APPROVED** [review]

**Pull request merged**